sdk/iot/Azure.ResourceManager.IotFirmwareDefense/api/Azure.ResourceManager.IotFirmwareDefense.net8.0.cs

```diff
@@ -113,15 +123,7 @@ protected IotFirmwareCollection() { }
     public partial class IotFirmwareData : Azure.ResourceManager.Models.ResourceData, System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.IotFirmwareDefense.IotFirmwareData>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.IotFirmwareDefense.IotFirmwareData>
     {
         public IotFirmwareData() { }
-        public string Description { get { throw null; } set { } }
-        public string FileName { get { throw null; } set { } }
-        public long? FileSize { get { throw null; } set { } }
-        public string Model { get { throw null; } set { } }
-        public Azure.ResourceManager.IotFirmwareDefense.Models.FirmwareProvisioningState? ProvisioningState { get { throw null; } }
-        public Azure.ResourceManager.IotFirmwareDefense.Models.FirmwareAnalysisStatus? Status { get { throw null; } set { } }
-        public System.Collections.Generic.IList<Azure.ResourceManager.IotFirmwareDefense.Models.FirmwareAnalysisStatusMessage> StatusMessages { get { throw null; } }
-        public string Vendor { get { throw null; } set { } }
-        public string Version { get { throw null; } set { } }
+        public Azure.ResourceManager.IotFirmwareDefense.Models.FirmwareProperties Properties { get { throw null; } set { } }
```

Comment:
this should be a lot of breaking change.
This is because in your new swagger, the `properties` property of this model does not have the `x-ms-client-flatten` flag, but the previous version has it.
To resolve this, we need to add a directive like this:
```
directive:
   - from: swagger-document
     where: $.definitions.IotFirmWare # find its real path in your swagger
     transform: $.properties.properties["x-ms-client-flatten"] = true
```
BTW there would be breaking changes for other languages' libraries as well.
If possible, I would recommend you to add it directly to your spec.

Reply:
Considering this RP is migrated from swagger to typespec, we should not reapply the flatten - instead, we should do this:
Introduce a partial class of this type, and put it in `src/Customization` directory like this:
```
namespace Azure.ResourceManager.IotFirmwareDefense;

public partial class IotFirmwareData
{
	[EditorBrowsable(EditorBrowsableState.Never)]
	public string Description
	{
		get => Properties?.Description;
		set {
			if (Properties is null)
			{
				Properties = new();
			}
			Properties.Description = value;
		}
	}
}
```